libvast/src/system/configuration.cpp

```diff
@@ -52,15 +52,20 @@ void initialize_factories() {
 
 configuration::configuration() {
   detail::add_message_types(*this);
-  // Use 'vast.conf' instead of generic 'caf-application.ini' and fall back to
-  // $PREFIX/etc/vast if no local vast.conf exists.
-  if ((path::current() / "vast.conf").is_regular_file()) {
-    config_file_path = "vast.conf";
-  } else {
-    auto global_conf = path{VAST_INSTALL_PREFIX} / "etc" / "vast" / "vast.conf";
-    if (!global_conf.is_regular_file())
-      global_conf = path{"/etc/vast/vast.conf"};
-    config_file_path = global_conf.str();
+  // Use 'vast.conf' instead of generic 'caf-application.ini'.
+  auto config_path_candidates = std::vector{
+    path::current() / "vast.conf",
+  };
+  auto binary = detail::objectpath();
+  if (binary)
+    config_path_candidates.push_back(binary->parent().parent() / "share"
+                                     / "vast" / "schema");
+  config_path_candidates.emplace_back("/etc/vast/vast.conf");
+  for (auto& p : config_path_candidates) {
```

Comment:
```suggestion
  for (const auto& p : config_path_candidates) {
```

Reply:
```
$ git grep 'for (auto&' | wc
    207    1368   13621

$ git grep 'for (const auto&' | wc
     22     171    1568
```
In principle I'm in favor of this suggestion, but I think it is a question of revising the coding guidelines.